modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/display/context/AssetCategoriesNavigationDisplayContext.java

```diff
@@ -166,34 +157,37 @@ public List<KeyValuePair> getAvailableVocabularyNames() {
 		Set<Long> availableAssetVocabularyIdsSet = SetUtil.fromArray(
 			getAvailableAssetVocabularyIds());
 
-		Stream<Long> availableAssetVocabularyIdsStream =
-			availableAssetVocabularyIdsSet.stream();
-
-		return availableAssetVocabularyIdsStream.filter(
-			assetVocabularyId ->
-				Arrays.binarySearch(assetVocabularyIds, assetVocabularyId) < 0
-		).map(
-			AssetVocabularyLocalServiceUtil::fetchAssetVocabulary
-		).map(
-			this::_toKeyValuePair
-		).sorted(
-			new KeyValuePairComparator(false, true)
-		).collect(
-			Collectors.toList()
-		);
+		List<KeyValuePair> vocabularyNames = TransformUtil.transform(
+			availableAssetVocabularyIdsSet,
+			assetVocabularyId -> {
+				int index = Arrays.binarySearch(
+					assetVocabularyIds, assetVocabularyId);
+
+				if (index >= 0) {
+					return null;
+				}
+
+				return _toKeyValuePair(
+					AssetVocabularyLocalServiceUtil.fetchAssetVocabulary(
+						assetVocabularyId));
+			});
+
+		vocabularyNames.sort(new KeyValuePairComparator(false, true));
+
+		return vocabularyNames;
 	}
 
 	public List<KeyValuePair> getCurrentVocabularyNames() {
-		LongStream longStream = Arrays.stream(getAssetVocabularyIds());
-
-		return longStream.boxed(
-		).map(
-			AssetVocabularyLocalServiceUtil::fetchAssetVocabulary
-		).map(
-			this::_toKeyValuePair
-		).collect(
-			Collectors.toList()
-		);
+		List<KeyValuePair> vocabularyNames = new ArrayList<>();
+
+		for (long assetVocabularyId : getAssetVocabularyIds()) {
```

Comment:
@julschong getAssetVocabularyIds() and getAssetVocabularies() should have consistent content based on the name. But currently they are not. We should make them consistent, then use getAssetVocabularies() here, then you can use TransformUtil rather than the for loop.